src/platform/plugins/shared/esql/public/triggers/esql_controls/control_flyout/identifier_control_form.tsx

```diff
@@ -62,23 +66,38 @@ export function FieldControlForm({
   onCancelControl,
   search,
   closeFlyout,
-}: FieldControlFormProps) {
+}: IdentifierControlFormProps) {
   const suggestedVariableName = useMemo(() => {
     const existingVariables = esqlVariables.filter((variable) => variable.type === variableType);
 
-    return initialState
-      ? `${initialState.variableName}`
-      : getRecurrentVariableName(
-          'field',
-          existingVariables.map((variable) => variable.key)
-        );
+    if (initialState) {
+      return initialState.variableName;
+    }
+
+    if (variableType === ESQLVariableType.FIELDS) {
+      return getRecurrentVariableName(
+        'field',
+        existingVariables.map((variable) => variable.key)
+      );
+    }
+
+    if (variableType === ESQLVariableType.FUNCTIONS) {
+      return getRecurrentVariableName(
+        'function',
+        existingVariables.map((variable) => variable.key)
+      );
+    }
+    return getRecurrentVariableName(
+      'variable',
+      existingVariables.map((variable) => variable.key)
+    );
```

Comment:
WDYT about getting a variable prefix and removing `if` conditions above? 

```suggestion
   const getVariablePrefix = (variableType: ESQLVariableType) => {...}
   return getRecurrentVariableName(getVariablePrefix(variableType), existingVariables);
```

Reply:
Sure done https://github.com/elastic/kibana/commit/7bb3bb14142baf4fbd9820d7feb558c6a292742a